src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3326,31 +3326,119 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
+                    double right_value = ASR::down_cast<ASR::RealConstant_t>(
+                                            ASRUtils::expr_value(rhs))->m_r;
+                    double result;
+                    switch (op) {
+                        case (ASR::logicalbinopType::And): {
+                            result = left_value == 0 ? left_value : right_value;
+                            break;
+                        }
+                        case (ASR::logicalbinopType::Or): {
+                            result = left_value != 0 ? left_value : right_value;
+                            break;
+                        }
+                        default : {
+                            throw SemanticError("Boolean operator type not supported",
+                                x.base.base.loc);
+                        }
+                    }
+                    value = ASR::down_cast<ASR::expr_t>(ASR::make_RealConstant_t(
+                        al, x.base.base.loc, result, dest_type));
+                    break;
                 }
+                case ASR::ttypeType::Character: {
+                    char* left_value = ASR::down_cast<ASR::StringConstant_t>(
+                                        ASRUtils::expr_value(lhs))->m_s;
+                    char* right_value = ASR::down_cast<ASR::StringConstant_t>(
+                                            ASRUtils::expr_value(rhs))->m_s;
+                    char* result;
+                    switch (op) {
+                        case (ASR::logicalbinopType::And): {
+                            result = std::strcmp(left_value, "") == 0 ? left_value : right_value;
+                            break;
+                        }
+                        case (ASR::logicalbinopType::Or): {
+                            result = std::strcmp(left_value, "") != 0 ? left_value : right_value;
+                            break;
+                        }
+                        default : {
+                            throw SemanticError("Boolean operator type not supported",
+                                x.base.base.loc);
+                        }
+                    }
+                    value = ASR::down_cast<ASR::expr_t>(ASR::make_StringConstant_t(
+                        al, x.base.base.loc, result, dest_type));
+                    break;
+                }
+
+                default:
+                    throw SemanticError("Boolean operation not supported on objects of type '"
+                            + ASRUtils::type_to_str_python(dest_type) + "'",
+                                x.base.base.loc);
             }
-            value = ASR::down_cast<ASR::expr_t>(ASR::make_LogicalConstant_t(
-                al, x.base.base.loc, result, dest_type));
+
+            tmp = ASR::make_LogicalBinOp_t(al, x.base.base.loc, lhs, op, rhs, dest_type, value);
         }
-        tmp = ASR::make_LogicalBinOp_t(al, x.base.base.loc, lhs, op, rhs, dest_type, value);
```

Comment:
I think the original place for constructing `tmp` is correct. `tmp` should be constructed even when the compile-time value is not available.

Reply:
Done!

Reply:
It does not seem to be updated for me. Can you construct tmp at the original/previous place it was being constructed? I think/guess that could help fix the issue that you are facing.

Reply:
I am pushing the changes.